internal/internal_workflow_testsuite.go

```diff
@@ -2363,6 +2367,37 @@ func (env *testWorkflowEnvironmentImpl) ExecuteNexusOperation(params executeNexu
 	}
 	env.runningNexusOperations[seq] = handle
 
+	var opID string
+	if params.options.ScheduleToCloseTimeout > 0 {
+		// Timer to fail the nexus operation due to schedule to close timeout.
```

Comment:
Hrmm, do we usually mimic timeouts like these in test suites? How does activity execution work in the test suite with regards to schedule to close timeout? Everything LGTM and will mark approved, just want to make sure this is normal for our test suite.

Reply:
Looks like some timeouts are broken in the test env (according to @rodrigozhou who looked into it). AFAIC, this is part of the behavior for nexus operations. It's enforced in the real server and the Java test server, don't see a good reason not to enforce it here.

Reply:
I think we need to understand/confirm how timeouts are expected to work in the Go test suite for external calls. I admit being unfamiliar. Do activities in the test suite respect their timeouts? (they might, I just want to check)

Reply:
When an activity is run by the workflow test environment (not mocked) we do propagate the timeout in the activities environment and rely on the activity to return to fail the operation which isn't how the real server works of course.

Reply:
:+1: Works for me